src/meta/client/MetaClient.h

```diff
@@ -35,35 +35,60 @@ using TagSchemas = std::unordered_map<std::pair<TagID, SchemaVer>,
 using EdgeSchemas = std::unordered_map<std::pair<EdgeType, SchemaVer>,
                                        std::shared_ptr<const SchemaProviderIf>>;
 
-using TagIndexes = std::unordered_map<TagIndexID, std::shared_ptr<const cpp2::IndexProperties>>;
-using EdgeIndexes = std::unordered_map<EdgeIndexID, std::shared_ptr<const cpp2::IndexProperties>>;
+using Indexes = std::unordered_map<IndexID, std::shared_ptr<cpp2::IndexFields>>;
+
+// IndexID => TagID
+using IndexTagMap  = std::unordered_map<IndexID, TagID>;
+
+// IndexID => EdgeType
+using IndexEdgeMap = std::unordered_map<IndexID, EdgeType>;
+
+using TagIndexPair  = std::pair<TagID, cpp2::IndexFields>;
+using EdgeIndexPair = std::pair<EdgeType, cpp2::IndexFields>;
+
+using TagIndexTuple  = std::tuple<TagID, IndexID, cpp2::IndexFields>;
+using EdgeIndexTuple = std::tuple<EdgeType, IndexID, cpp2::IndexFields>;
```

Comment:
A suggestion, Is it better to use `IndexItem` directly?

Reply:
`IndexItem` not include the `tag id` and `edge type` ?

Reply:
TagId and EdgeType is required. 
In pr1360,  @dangleptr suggested that I change it as below , so could you help unify it as below 😊?
```
struct IndexItem {
     1: required IndexID          index_id,
     2: required i32              tagOrEdge,
     3: required list<ColumnDef>  cols,
}
```

Reply:
Let me have a try.